workflow/controller/operator.go

```diff
@@ -975,7 +975,9 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 
 			// If the node is pending and the pod does not exist, it could be the case that we want to try to submit it
 			// again instead of marking it as an error. Check if that's the case.
-			if node.Pending() {
+			// Added `ContainerCreating` condition to avoid workflow stuck in running state with the pod deleted in ContainerCreating
+			// https://github.com/argoproj/argo-workflows/issues/5543
+			if node.Pending() && !strings.Contains(node.Message, "ContainerCreating") {
```

Comment:
Why is the node not pending if the pod is creating? or is the node running? I wonder if something is wrong with `inferPodPhase`, take a look at this:

https://github.com/argoproj-labs/argo-dataflow/blob/main/controllers/infer.go#L40

Reply:
I think a pod can have `phase=Running`, but has containers are in `ContainerCreating` status. 

I think a deep-dive into pod status?